src/main/java/io/jenkins/plugins/coverage/source/DefaultSourceFileResolver.java

```diff
@@ -231,6 +234,32 @@ private FilePath tryFindSourceFile(File workspace) {
                 }
             }
 
+            // last but not least, search for the sourceFilePath within the entire workspace
+            // for Cobertura, this step attempts to ignore CoverageFeatureConstants.FEATURE_SOURCE_FILE_PATH
+            // or 'source-file-path':'coverage/sources/source' as set in Cobertura output XML files
+            /* NOTES:
+                https://issues.jenkins.io/browse/JENKINS-5235
+                https://github.com/jenkinsci/cobertura-plugin/issues/103
+                https://github.com/jenkinsci/cobertura-plugin/issues/61
+
+            */
+            try (Stream<Path> walk = Files.walk(Paths.get(workspace.getAbsolutePath()))) {
```

Comment:
We have a gigantic project an the performance isn't bad.  A single build is over 2 hours without Cobertura or Code Coverage, I haven't noticed any less performance of using this in Code Coverage vs using Cobertura plugin directly.  Of course, "for our use".

Reply:
The issue is that Cobertura paths are fully qualified, so if the job's name is changed, or it's moved to folders, or the job is cloned to another Jenkins which may be in another directory structure, the section fails.  The code knows the workspace URI and the sourcefile subpath; though the workspace may be a different URI, so may the subpath and betwixt those two the URI may be different than in the Cobertura file.
The significance of all this is that this plugin is useful for comparing a current build with past builds, and upon URI changes, that history is lost.  We added the above code to our own version to keep the history.
This section is last because in an environment in which none of the situations above has happened, this section is ignored; it's the last resort.   Also, on average, only 50% of the file system will be walked to find a match (this is similarly done within databases on records to find a specific record). 
The only other option for us was to parse Covertura files and update the path within; that requires xml parsing and for us that's nearly 3000 files to update, though that may not happen often, this section of code removes the need from the user.
If Cobertura would update to use relative paths, then this code section wouldn't be necessary :) . But my understanding is that it can't or won't be done any time soon, unfortunately.

Possibly, a user provided boolean may help here. e.g. if the user doesn't care, do this section. If the user prefers to have the non-found files removed from the history, then select falsey.